src/tx_xform_prv.erl

```diff
@@ -101,14 +103,30 @@ pp(Forms, F) ->
 
 clauses(Found, L0) ->
     {Cs, _} = lists:mapfoldl(
-                fun({Mod, _, {RecName, _}}, L) ->
+                fun({Mod, _, {RecName, _}, _}, L) ->
                         {{clause, L, [{record, L, RecName, []}],
                           [],
                           [{atom, L+1, Mod}]},
                          L+2}
                 end, L0, Found),
     Cs.
 
+type_clauses(Found, L0) ->
+    {Cs, _} = lists:mapfoldl(
+                fun({Mod, _, _, Str}, L) ->
+                  case Str of
+                    {error, _} = Err ->
+                        Err;
```

Comment:
But you're in a `mapfoldl`, so the proper return value would be `{Err, L}`, right?